src/Microsoft.XmlSerializer.Generator/src/Sgen.cs

```diff
@@ -565,9 +565,14 @@ private static Assembly SgenAssemblyResolver(object source, ResolveEventArgs e)
                     return null;
                 }
 
-                if(s_referencedic.ContainsKey(assemblyname))
+                if (s_referencedic.ContainsKey(assemblyname))
                 {
                     string reference = s_referencedic[assemblyname];
+
+                    //for System.ServiceModel.Primitives, we need to load its execution assembly rather than reference assembly
+                    if (assemblyname.Equals("System.ServiceModel.Primitives"))
```

Comment:
who's providing this assembly name? couldn't they pass the path themselves? i.e. AssemblyName provides Codebase for that (see: https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assemblyname?view=netcore-2.1)

Reply:
This code is a custom Assembly resolver, the whole point is that the CLR needs to load the assembly and it doesn't know where to find it. If it did know where to find it, it wouldn't be calling this event handler. So the AssemblyName is coming from the CLR coming across a type which lives in that assembly. In other words, the IL in an assembly references a type and says what assembly that type was found in at compile time and the CLR can't find it so is asking our handler.

---
In reply to: [220474266](https://github.com/dotnet/corefx/pull/32403#discussion_r220474266) [](ancestors = 220474266)